api/task_check_run.go

```diff
@@ -77,8 +77,15 @@ const (
 	TaskCheckDatabaseConnect TaskCheckType = "bb.task-check.database.connect"
 	// TaskCheckInstanceMigrationSchema is the task check type for migrating schemas.
 	TaskCheckInstanceMigrationSchema TaskCheckType = "bb.task-check.instance.migration-schema"
+	// TaskCheckGeneralEarliestAllowedTime is the task check type for earliest allowed time.
+	TaskCheckGeneralEarliestAllowedTime TaskCheckType = "bb.task-check.general.earliest-allowed-time"
```

Comment:
Per offline discussion, let's update notBeforeTs to earliestAllowedTs.  Negative term is not that readable.